_posts/2023-02-06-tesla-ypp-2022-blogpost.md

```diff
@@ -0,0 +1,245 @@
+
+We've used GitHub and GitHub actions combined with Terraform, Helm, and Docker for deploying and CI/CD.
+
+In the context of testing, we've used Postman to test REST and Gatling for performance testing.
+
+## ROCK PAPER SCISSORS ARCHITECTURE
+
+<img src="{{ '/img/2023-02-06-tesla-ypp-2022-blogpost/rock-paper-scissors-architecture.png' | prepend: site.baseurl }}" alt="It shows every resource used to create the whole rock paper scissors application" class="image fit" style="margin:0px auto; max-width:70%; height:70%">
+
+A user will first need to authenticate themselves using Cognito before sending requests.
+
+An API Gateway was not necessary but was added for educational purposes.
+As a result, we needed to link it to our RPS(Rock-Paper-Scissors)-application via a VPC link to our Network Load Balancer (NLB) next to an Application Load Balancer (ALB) that will be connected to our RPS application.
+
+The RPS application then communicates with our Proxy-API that will (based on the active profile) communicate with one of the Tesla-APIs.
+
+Finally, the key rotation mentioned in the previous section is also included in the diagram.
+
+## GOOGLE HOME ARCHITECTURE
+
+<img src="{{ '/img/2023-02-06-tesla-ypp-2022-blogpost/google-architecture.png' | prepend: site.baseurl }}" alt="It shows every resource used to create the whole google application" class="image fit" style="margin:0px auto; max-width:70%; height:70%">
+
+The applications are deployed on Kubernetes and made accessible through Ingress and Route53.
+The Google box is linked to a Google Home account, which, in turn, is connected to our applications.
+The Proxy API is communicating with the Tesla APIs
+
+## Used Technologies
+
+The development of our project required the use of a variety of technologies to ensure a successful outcome.
+Here's a brief overview of each technology and why it was chosen.
+
+1. AWS: AWS was chosen as the cloud platform for our project because of its scalability, efficiency, and security benefits.
+By leveraging the resources offered by AWS, we were able to take advantage of the many benefits of cloud computing.
+
+2. Terraform: Terraform is an open-source tool used for infrastructure as code.
+This tool allowed us to automate the provisioning and management of our infrastructure on AWS, making it easier for us to manage and maintain our infrastructure over time.
+The whole infrastructure will be deployed on Monday in the morning and destroyed on Friday night using a cronjob.
+
+3. GitHub + GitHub Actions: GitHub is a version control platform that we used to store and manage our code.
+GitHub Actions is a continuous integration and deployment (CI/CD) platform that allowed us to automate the build, test, and deployment of our application.
+The CI/CD will run every time a pull request is made.
+This to validate that the application can be build using maven.
+Another trigger for the CI/CD to run is when code is pushed to the develop branch.
```

Comment:
Explain the workflow more clearly. 
PR created -> trigger
push to develop -> trigger? (what to where?)